### ISSUE TITLE
Not getting new messages after Redis restart

### ISSUE THREAD

**Issue by nalexeric** [mpneuried/rsmq-worker#30]
**Background:**

I have two services, one emitting messages, using `rsmq.sendMessage({qname: 'cartName', message: msg})` (after making sure the queue with name 'cartName' exists).

In the other service I'm using RSMQWorker to connect to the same Redis host / queue using the following config: 

```javascript
RSMQWorker = require('rsmq-worker');
let worker = new RSMQWorker('cartQueue', {
    interval: [0, 1],
    invisibletime: 30,
    maxReceiveCount: 1,
    redisPrefix: ns,
    host: host,
    port: 32768,
    autostart: true,
    timeout: 10
});
```

Everything works as expected, except for the cases when Redis crashes. 
If I stop the Redis server while both services are running and restart it after a while, all the messages sent from the first service while Redis was down will be processed on the first server (sendMessage works without errors). However they won't reach the worker service, unless I manually restart the second service. 

So basically, only the sender service reconnects to Redis in case of a crash, while the worker service doesn't. 

Is this a normal behaviour? Is there anything in the config of the worker service that prevents it from reconnecting to Redis in case of a crash (I've tried tweaking some of the options but to no effect)?

**Comment by mpneuried** [comment]
I will habe a look at it soon ;-)

**Comment by mpneuried** [comment]
hi, i created a test case [_src/test/pingpong.coffee](../blob/master/_src/test/pingpong.coffee) to check teh behaviuor of connect and reconnection.

With this i was able to find and fix your case ;-)

**Comment by jost-s** [comment]
@mpneuried  Hi, I'm having the same issue using v0.5.2.
When the redis client is disconnected for some reason, it will reconnect automatically. However, the worker doesn't pick up new messages after the reconnect, I have to restart the whole service (redis client and worker).
